drivers/pds/lib/utils.go

```diff
@@ -0,0 +1,578 @@
+
+				if err != nil {
+					logrus.Warnf("An Error Occured while creating deployment %v", err)
+					return deploymentsMap, err
+				}
+				err = ValidateDataServiceDeployment(deployment)
+				if err != nil {
+					return deploymentsMap, err
+				}
+				deploymentsMap[ds] = append(deploymentsMap[ds], deployment)
+			}
+		}
+	}
+	return deploymentsMap, nil
+}
+
+//GetAllSupportedDataServices get the supported datasservices and returns the map
+func GetAllSupportedDataServices() map[string]string {
+	dataService, _ := components.DataService.ListDataServices()
+	for _, ds := range dataService {
+		if !*ds.ComingSoon {
+			dataServiceNameIDMap[ds.GetName()] = ds.GetId()
+		}
+	}
+	for key, value := range dataServiceNameIDMap {
+		logrus.Infof("dsKey %v dsValue %v", key, value)
+	}
+	return dataServiceNameIDMap
+}
+
+//ValidateDataServiceVolumes validates the volumes
+func ValidateDataServiceVolumes(deployment *pds.ModelsDeployment, dataService string, dataServiceDefaultResourceTemplateIDMap map[string]string, storageTemplateID string) (ResourceSettingTemplate, StorageOptions, StorageClassConfig) {
+	var config StorageClassConfig
+	ss, err := k8sApps.GetStatefulSet(deployment.GetClusterResourceName(), GetAndExpectStringEnvVar("NAMESPACE"))
+	if err != nil {
+		logrus.Warnf("An Error Occured while getting statefulsets %v", err)
+	}
+	err = k8sApps.ValidatePVCsForStatefulSet(ss, timeOut, timeInterval)
+	if err != nil {
+		logrus.Errorf("An error occured while validating pvcs of statefulsets %v ", err)
+	}
+	pvcList, err := k8sApps.GetPVCsForStatefulSet(ss)
+	if err != nil {
+		logrus.Warnf("An Error Occured while getting pvcs of statefulsets %v", err)
+	}
+
+	for _, pvc := range pvcList.Items {
+		sc, err := k8sCore.GetStorageClassForPVC(&pvc)
+		if err != nil {
+			logrus.Errorf("Error Occured while getting storage class for pvc %v", err)
+		}
+		scAnnotation := sc.Annotations
+		for k, v := range scAnnotation {
+			if k == "kubectl.kubernetes.io/last-applied-configuration" {
+				logrus.Infof("Storage Options Values %v", v)
+				data := []byte(v)
+				err := json.Unmarshal(data, &config)
+				if err != nil {
+					logrus.Errorf("Error Occured while getting volume params %v", err)
+				}
+			}
+		}
+	}
+
+	var resourceTemp ResourceSettingTemplate
+	var storageOp StorageOptions
+
+	rt, err := components.ResourceSettingsTemplate.GetTemplate(dataServiceDefaultResourceTemplateIDMap[dataService])
+	if err != nil {
+		logrus.Errorf("Error Occured while getting resource setting template %v", err)
+	}
+	resourceTemp.Resources.Requests.CPU = *rt.CpuRequest
+	resourceTemp.Resources.Requests.Memory = *rt.MemoryRequest
+	resourceTemp.Resources.Requests.Storage = *rt.StorageRequest
+	resourceTemp.Resources.Limits.CPU = *rt.CpuLimit
+	resourceTemp.Resources.Limits.Memory = *rt.MemoryLimit
+
+	st, err := components.StorageSettingsTemplate.GetTemplate(storageTemplateID)
+	if err != nil {
+		logrus.Errorf("Error Occured while getting storage template %v", err)
```

Comment:
return the error, else st might be null and it will result in panic

Reply:
done